src/nbl/video/alloc/SimpleGPUBufferAllocator.cpp

```diff
@@ -16,5 +16,15 @@ SimpleGPUBufferAllocator::value_type SimpleGPUBufferAllocator::allocate(size_t b
     auto reqs = mBufferMemReqs;
     reqs.vulkanReqs.size = bytes;
     reqs.vulkanReqs.alignment = alignment;
-    return mDriver->createGPUBufferOnDedMem(reqs,canUpdateViaCmdBuff);
+
+    // Todo(achal): Need to get these values from the user somehow
+    // This doesn't matter right now because updateBufferRangeViaStagingBuffer 
+    // only makes use of upstreaming buffer which would be TRANSFER_SRC
+    IGPUBuffer::SCreationParams creationParams = {};
+    creationParams.usage = IGPUBuffer::EUF_TRANSFER_SRC_BIT;
+    creationParams.sharingMode = asset::ESM_EXCLUSIVE;
+    creationParams.queueFamilyIndexCount = 0u;
+    creationParams.queueFamilyIndices = nullptr;
+
+    return mDriver->createGPUBufferOnDedMem(creationParams, reqs, canUpdateViaCmdBuff);
```

Comment:
@achalpandeyy could we fold `canUpdateViaCmdBuff` into `IGPUBuffer::ScreationParams` ?

Reply:
ok lets do it